src/sgl/device/tests/test_reflection.py

```diff
@@ -669,5 +669,309 @@ def test_get_type_layout(test_id: str, device_type: sgl.DeviceType):
+        module_name=f"module_from_source_{test_id}",
+        source=r"""
+        int foo(float3 val)
+        {
+            return 0;
+        }
+    """,
+    )
+
+    # Find and verify the generic function.
+    f1 = module.layout.find_function_by_name("foo")
+    assert f1 is not None
+    assert not f1.is_overloaded
+    assert len(f1.overloads) == 0
+    assert len(f1.parameters) == 1
+    assert f1.parameters[0].name == "val"
+    assert f1.parameters[0].type.full_name == "vector<float,3>"
+
+    # Specialize the function with int.
+    f2 = f1.specialize_with_arg_types([module.layout.find_type_by_name("float3")])
+    assert f2 is not None
+    assert not f2.is_overloaded
+    assert len(f2.overloads) == 0
+    assert len(f2.parameters) == 1
+    assert f2.parameters[0].name == "val"
+    assert f2.parameters[0].type.full_name == "vector<float,3>"
+
+
+@pytest.mark.skip(reason="Crashes slang")
+@pytest.mark.parametrize("device_type", helpers.DEFAULT_DEVICE_TYPES)
+def test_specialize_interface_function(test_id: str, device_type: sgl.DeviceType):
+    device = helpers.get_device(type=device_type)
+
+    # Create a session, and within it a module.
+    session = helpers.create_session(device, {})
+    module = session.load_module_from_source(
+        module_name=f"module_from_source_{test_id}",
+        source=r"""
+        int foo(IFloat val)
+        {
+            return 0;
+        }
+    """,
+    )
+
+    # Find and verify the generic function.
+    f1 = module.layout.find_function_by_name("foo")
+    assert f1 is not None
+    assert not f1.is_overloaded
+    assert len(f1.overloads) == 0
+    assert len(f1.parameters) == 1
+    assert f1.parameters[0].name == "val"
+    assert f1.parameters[0].type.name == "IFloat"
+
+    # Specialize the function with int.
+    f2 = f1.specialize_with_arg_types([module.layout.find_type_by_name("float")])
+    assert f2 is not None
+    assert not f2.is_overloaded
+    assert len(f2.overloads) == 0
+    assert len(f2.parameters) == 1
+    assert f2.parameters[0].name == "val"
+    assert f2.parameters[0].type.full_name == "float"
+
+
+@pytest.mark.parametrize("device_type", helpers.DEFAULT_DEVICE_TYPES)
+def test_specialize_generic_function(test_id: str, device_type: sgl.DeviceType):
+    device = helpers.get_device(type=device_type)
+
+    # Create a session, and within it a module.
+    session = helpers.create_session(device, {})
+    module = session.load_module_from_source(
+        module_name=f"module_from_source_{test_id}",
+        source=r"""
+        int foo<T>(T val)
+        {
+            return 0;
+        }
+    """,
+    )
+
+    # Find and verify the generic function.
+    f1 = module.layout.find_function_by_name("foo")
+    assert f1 is not None
+    assert not f1.is_overloaded
+    assert len(f1.overloads) == 0
+    assert len(f1.parameters) == 1
+    assert f1.parameters[0].name == "val"
+    assert f1.parameters[0].type.name == "T"
+
+    # Specialize the function with int.
```

Comment:
```suggestion
    # Specialize the function with float3.
```

or just `Specialize the function`.
Same in all the other tests.